src/App.js

```diff
@@ -1,53 +1,60 @@
-import React from "react";
+import React, { useState } from "react";
 import { Switch, Redirect, Route } from "react-router-dom";
 import AuthenticationRoute from "./containers/AuthenticationRoute";
 import routes from "./helpers/routes";
 import LoginScreen from "./components/LoginScreen";
 import Currency from "./components/Currency";
+import GlobalErrContext from './context/GlobalErrContext';
 
 const App = () => {
+  const [globalError, setGlobalErr] = useState(null);
+
   return (
-    <Switch>
-      {/* ============== START OF auth insensitive routes ============== */}
-      <Route
-        path="/currency"
-        component={() => (
-          <Currency
-            value="1500015646"
-            withCurrencyLabel={true}
-            withSymbol={true}
-            currency=""
-            fractionDigits="3"
-          />
-        )}
-      />
-      {/* ============== END OF auth insensitive routes ================ */}
+    <GlobalErrContext.Provider value={[
```

Comment:
Take a look how Mushegh did this part. He exports AuthProvider from the hook, so that there's no need to do this